src/peft/tuners/lora/config.py

```diff
@@ -88,10 +88,11 @@ class LoraConfig(PeftConfig):
     layers_to_transform: Optional[Union[List[int], int]] = field(
         default=None,
         metadata={
-            "help": "The layer indexes to transform, is this argument is specified, PEFT will transform only the layers indexes that are specified inside this list. If a single integer is passed, PEFT will transform only the layer at this index."
+            "help": "The layer indexes to transform, is this argument is specified, PEFT will transform only the layers indexes that are specified inside this list. If a single integer is passed, PEFT will transform only the layer at this index. "
+            "This cannot be used when `target_modules` is a regex expression"
```

Comment:
How about, instead of wording this in the negative ("cannot be used"), we word it positively: "This only works when `target_modules` is a list of str." or something like that. Should make it easier for users to know what they need to do.